bigml/api_handlers/resourcehandler.py

```diff
@@ -517,8 +517,44 @@ def get_resource(self, resource, **kwargs):
         resource_id = get_resource_id(resource)
 
         if resource_id:
-            return self._get("%s%s" % (self.url, resource_id),
-                             **kwargs)
+            return self._get("%s%s" % (self.url, resource_id), **kwargs)
+
+    def update_resource(self, resource, changes, **kwargs):
+        """Updates a remote resource.
+
+           The resource parameter should be a string containing the
+           resource id or the dict returned by the corresponding create method.
+
+        """
+        resource_id, error = self.final_resource(resource)
+        if error or resource_id is None:
+            raise Exception("Failed to update %s. Only correctly finished "
+                            "resources can be updated. Please, check "
+                            "the resource status.")
+        body = json.dumps(changes)
+        return self._update("%s%s" % (self.url, resource_id), body, **kwargs)
+
+    def delete_resource(self, resource, **kwargs):
+        """Delete a remote resource
+
+        """
+        resource_id = get_resource_id(resource)
+        if resource_id:
+            return self._delete("%s%s" % (self.url, resource_id), **kwargs)
+
+    def _download_resource(self, resource, filenamen, retries=10):
+        """Download CSV information from downloadable resources
+
+        """
```

Comment:
you're not documenting the return value of any of these three functions... i guess it's more or less obvious, and these are not really part of the public interface, so it's okay not to launch another test run if this is my only comment.